Add firehose reloaded; remove dataflow

max-standard opened the pull request:
This pull request changes Hedwig firehose facility implementation from Dataflow to Pub/Sub Cloud Storage subscription—see https://cloud.google.com/pubsub/docs/cloudstorage page.

**Done:** After I merge https://github.com/cloudchacho/terraform-google-hedwig-topic/pull/2 pull request, then I should also bump the https://github.com/cloudchacho/terraform-google-hedwig/blob/bae78b0aac0d135aaea1edafe6707fa269aa1f54/topics.tf#L1-L5 version attribute because **Guess:** My https://github.com/cloudchacho/terraform-google-hedwig-topic/pull/2 change should cause a major version bump since I'm removing so many dataflow input variables—see https://github.com/cloudchacho/terraform-google-hedwig/pull/4/commits/9ff62c5f81dd962bc26ac4ceabf28802785bf6d5 commit.

max-standard committed:
Add firehose reloaded; remove dataflow

max-standard committed:
Make max_duration and max_bytes configurable

max-standard committed:
Add dynamic avro_config

shsamkit commented on variables.tf:
```diff
@@ -113,9 +82,24 @@ variable "topics" {
     enable_firehose = optional(bool)
 
+    # Variable firehose_bucket declares the bucket for firehose—that is, saved—messages (should already exist)
+    firehose_bucket = optional(string)
```
Same suggestion/q. from the other PR : https://github.com/cloudchacho/terraform-google-hedwig-topic/pull/2#discussion_r1417627468

max-standard replied:
See https://github.com/cloudchacho/terraform-google-hedwig/pull/4/commits/8b8ff6899c5faf8fb1d3acc9d1d8fed04d2d282f commit.

max-standard committed:
Consolidate firehose configs into firehose_config object

max-standard committed:
Make bucket field required

max-standard committed:
Typo: Reset hedwig-topic

max-standard committed:
Make firehose_config bucket field required

max-standard committed:
Remove enabled field

Note: Instead of enabled field, just add the firehose_config configuration object; the default is null; therefore, no firehose by default

max-standard committed:
Update hedwig-topic from >= 2.2 to >= 3.0
